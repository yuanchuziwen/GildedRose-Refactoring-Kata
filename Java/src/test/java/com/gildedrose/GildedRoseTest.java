package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class GildedRoseTest {

    @Test
    void foo() {
        Item[] items = new Item[] { new Item("foo", 0, 0) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        assertEquals("fixme", app.items[0].name);
    }

    @Test
    public void approvalTest() {
        Item[] items = genItems();

        GildedRose app = new GildedRose(items);

        int days = 2;

        for (int i = 0; i < days; i++) {
            System.out.println("-------- day " + i + " --------");
            for (Item item : items) {
                System.out.println(item);
            }
            System.out.println();
            app.updateQuality();
        }

        String res = collectRes(items);
        Approvals.verifyAll("", items);
    }

    private String collectRes(Item[] items) {
        StringBuilder sb = new StringBuilder();
        for (Item item : items) {
            sb.append(item.toString());
            sb.append("\n");
        }
        return sb.toString();
    }

    private static Item[] genItems() {
        Item[] items = new Item[]{
            new Item("+5 Dexterity Vest", 10, 20), //
            new Item("Aged Brie", 2, 0), //
            new Item("Elixir of the Mongoose", 5, 7), //
            new Item("Sulfuras, Hand of Ragnaros", 0, 80), //
            new Item("Sulfuras, Hand of Ragnaros", -1, 80),
            new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
            // this conjured item does not work properly yet
            new Item("Conjured Mana Cake", 3, 6),
            new Item("Aged Brie", 10, 30),
            new Item("Backstage passes to a TAFKAL80ETC concert", 12, 20),
            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 30),
            new Item("Backstage passes to a TAFKAL80ETC concert", 0, 30),
            new Item("Sulfuras, Hand of Ragnaros", 10, 80),
            new Item("Regular item", 10, 30),
            new Item("Regular item", 0, 30),
            new Item("Regular item", -1, 0),
            new Item("else", -1, 48),
            new Item("Aged Brie", -1, 49),
            new Item("Aged Brie", -1, 48)};
        return items;
    }

}
