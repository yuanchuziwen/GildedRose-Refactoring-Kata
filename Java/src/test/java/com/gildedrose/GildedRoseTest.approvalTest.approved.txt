[00] = +5 Dexterity Vest, 8, 18
[01] = Aged Brie, 0, 2
[02] = Elixir of the Mongoose, 3, 5
[03] = Sulfuras, Hand of Ragnaros, 0, 80
[04] = Sulfuras, Hand of Ragnaros, -1, 80
[05] = Backstage passes to a TAFKAL80ETC concert, 13, 22
[06] = Backstage passes to a TAFKAL80ETC concert, 8, 50
[07] = Backstage passes to a TAFKAL80ETC concert, 3, 50
[08] = Conjured Mana Cake, 1, 4
[09] = Aged Brie, 8, 32
[10] = Backstage passes to a TAFKAL80ETC concert, 10, 22
[11] = Backstage passes to a TAFKAL80ETC concert, 8, 50
[12] = Backstage passes to a TAFKAL80ETC concert, 3, 36
[13] = Backstage passes to a TAFKAL80ETC concert, -2, 0
[14] = Sulfuras, Hand of Ragnaros, 10, 80
[15] = Regular item, 8, 28
[16] = Regular item, -2, 26
[17] = Regular item, -3, 0
[18] = else, -3, 44
[19] = Aged Brie, -3, 50
[20] = Aged Brie, -3, 50
